Microsoft.WindowsAzure.Storage/src/blob_request_factory.cpp

```diff
@@ -535,6 +543,17 @@ namespace azure { namespace storage { namespace protocol {
         return request;
     }
 
+    web::http::http_request set_blob_tier(const utility::string_t& tier, const access_condition& condition, web::http::uri_builder& uri_builder, const std::chrono::seconds& timeout, operation_context context)
+    {
+        uri_builder.append_query(core::make_query_parameter(uri_query_component, component_tier, /* do_encoding */ false));
+        web::http::http_request request(base_request(web::http::methods::PUT, uri_builder, timeout, context));
+        
+        request.headers().add(ms_header_access_tier, tier);
+        add_append_condition(request, condition);
```

Comment:
why adding append condition? Isn't that only for append blob?

Reply:
Should be a mistake, will modify it.